File: shared/src/main/scala/com/melvic/ranker/models/Item.scala
package com.melvic.ranker.models

import java.time.LocalDateTime

final case class Item(
    id: Item.Id,
    name: String,
    description: String,
    score: Double,
    createdAt: LocalDateTime,
    lastUpdate: Update,
) extends HasTimestamp

object Item {
  type Id = Int

  def fromName(name: String): Item = {
    val creationDate = LocalDateTime.now
    Item(
      id = 1,
      name,
      description = "",
      score = 0.0,
      createdAt = creationDate,
      lastUpdate = Update.fromDate(creationDate)
    )
  }
}

File: src/main/java/com/melvic/ranker/items/Item.java
package com.melvic.ranker.items;

import com.melvic.ranker.topics.Topic;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.ManyToOne;
import java.time.LocalDateTime;

@Entity
public class Item {
    @Id
    private String id;
    private String name;
    private String description;

    @Column(updatable = false)
    @CreationTimestamp
    private LocalDateTime createdAt;

    @UpdateTimestamp
    private LocalDateTime lastModified;

    @ManyToOne
    private Topic topic;

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public LocalDateTime getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(LocalDateTime createdAt) {
        this.createdAt = createdAt;
    }

    public LocalDateTime getLastModified() {
        return lastModified;
    }

    public void setLastModified(LocalDateTime lastModified) {
        this.lastModified = lastModified;
    }

    public Topic getTopic() {
        return topic;
    }

    public void setTopic(Topic topic) {
        this.topic = topic;
    }
}
